File: API/Entities/OrderAggregate/Order.cs
namespace API.Entities.OrderAggregate
{
    public class Order
    {
         public int Id { get; set; }
        public string BuyerId { get; set; }
        public ShippingAddress ShippingAddress { get; set; }
        public string OrderDate { get; set; } = DateTime.Now.ToLongDateString();
        public List<OrderItem> OrderItems { get; set; }
        public long Subtotal { get; set; }
        public long DeliveryFee { get; set; }
        public OrderStatus OrderStatus { get; set; } = OrderStatus.Pending;
        public string PaymentIntentId { get; set; }

        public long GetTotal()
        {
            return Subtotal + DeliveryFee;
        }
    }
}

File: spa/src/app/models/order.ts
export interface Order {
    id: number
    buyerId: string
    shippingAddress: ShippingAddress
    orderDate: string
    orderItems: OrderItem[]
    subtotal: number
    deliveryFee: number
    orderStatus: string
    total: number
  }
  
  export interface ShippingAddress {
    fullName: string
    address1: string
    address2: string
    city: string
    country: string
    postalCode: string
    phone: string
  }
  
  export interface OrderItem {
    productId: number
    name: string
    pictureUrl: string
    price: number
    quantity: number
  }